platform/ios/test/MGLSourceTests.m

```diff
@@ -0,0 +1,24 @@
+#import "MGLMapViewTests.h"
+
+@interface MGLSourceTests : MGLMapViewTests
+
+@end
+
+@implementation MGLSourceTests
+
+- (void)testDuplicateSources {
+    MGLVectorSource *source1 = [[MGLVectorSource alloc] initWithIdentifier:@"my-source" URL:[NSURL URLWithString:@"mapbox://mapbox.mapbox-terrain-v2"]];
+    MGLVectorSource *source2 = [[MGLVectorSource alloc] initWithIdentifier:@"my-source" URL:[NSURL URLWithString:@"mapbox://mapbox.mapbox-terrain-v2"]];
+    
+    [self.mapView.style addSource: source1];
+    
+    @try {
```

Comment:
FYI, `@try`-`@catch` blocks are generally to be avoided in Objective-C code due to their poor performance. It’s technically necessary here because we’re trying to detect an exception being raised. However, the more idiomatic way to assert for the exception is to use the `XCTAssertThrows()`, `XCTAssertThrowsSpecific()`, or `XCTAssertThrowsSpecificNamed()` macro.